yarn-project/archiver/src/archiver/data_retrieval.ts

```diff
@@ -139,11 +140,53 @@ async function getBlockFromRollupTx(
   if (!allowedMethods.includes(functionName)) {
     throw new Error(`Unexpected method called ${functionName}`);
   }
-  const [headerHex, archiveRootHex, , , , bodyHex] = args! as readonly [Hex, Hex, Hex, Hex[], ViemSignature[], Hex];
+  // TODO(#9101): 'bodyHex' will be removed from below
+  const [headerHex, archiveRootHex, , , , bodyHex, blobInputs] = args! as readonly [
+    Hex,
+    Hex,
+    Hex,
+    Hex[],
+    ViemSignature[],
+    Hex,
+    Hex,
+  ];
 
   const header = Header.fromBuffer(Buffer.from(hexToBytes(headerHex)));
+  // TODO(#9101): Retreiving the block body from calldata is a temporary soln before we have
+  // either a beacon chain client or link to some blob store. Web2 is ok because we will
+  // verify the block body vs the blob as below.
   const blockBody = Body.fromBuffer(Buffer.from(hexToBytes(bodyHex)));
 
+  const blockFields = blockBody.toFields();
```

Comment:
@ TMNT team: for #9101, this `blockFields` will be the output of the blob source, and can be reconstructed into a block as below. The call will be simply `Body.fromFields(blockFields)` (currently logs are injected until #8954 turns logs into fields).